Alternative XQuery processor

adam-vessey opened the pull request.

adam-vessey committed:
Theoretical implementation...

adam-vessey committed:
BaseX stuff runs, huzzah.

adam-vessey committed:
Some coding standards fun.

adam-vessey committed:
Delete our new variables and preserve the processor on existing installations.

adam-vessey committed:
Add example script, and flesh out README.

adam-vessey committed:
Adjust script.

adam-vessey committed:
Set the executable.

adam-vessey committed:
Adjust comment.

adam-vessey committed:
Add typehints.

adam-vessey committed:
Full, default path.

willtp87 commented on includes/basex.inc:
```diff
@@ -0,0 +1,130 @@
+        '#type' => 'textfield',
+        '#title' => t('BaseX exectuable'),
+        '#description' => t('BaseX will be used to execute XQuery queries.<br/>!msg', array(
+          '!msg' => islandora_executable_available_message($executable))
```
110 | ERROR | [x] Closing parenthesis of array declaration must be
     |       |     on a new line

adam-vessey replied:
[Is copypasta](https://github.com/discoverygarden/islandora_xquery/blob/2deb7f281c1f7afb75ffcc74a5e339b4bc8d7e55/includes/admin.form.inc#L47)... shrug?

willtp87 commented on includes/basex.inc:
```diff
@@ -0,0 +1,130 @@
+<?php
+
+class IslandoraXqueryBaseX implements IslandoraXqueryConfigurableProcessorInterface {
```
3 | ERROR | [x] Missing class doc comment

willtp87 commented on includes/processor_admin.form.inc:
```diff
@@ -0,0 +1,81 @@
+
+  foreach ($form_state['storage']['implementations'] as $name => $class) {
+    $form['processor'][$name] = (is_callable(array($class, "getForm")) ?
+      call_user_func_array(array($class, "getForm"), array(array('processor', $name), &$form_state)) :
```
40 | ERROR | If the line declaring an array spans longer than 80
    |       | characters, each element should be broken into its own
    |       | line

willtp87 commented on includes/IslandoraXqueryNoExectuableException.inc:
```diff
@@ -0,0 +1,28 @@
+      $this->name,
+      $this->class,
+    );
+  }
```
27 | ERROR | [x] Expected 1 blank line after function; 0 found

willtp87 commented on includes/xquery.inc:
```diff
@@ -0,0 +1,59 @@
+  public static function getImplementations() {
+    module_load_include('inc', 'islandora_xquery', 'includes/zorba');
+    return array(
+      'zorba' => \Islandora\Zorba\Options::class,
```
33 | ERROR | [x] Namespaced classes/interfaces/traits should be
    |       |     referenced with use statements

adam-vessey committed:
Ensure our solr result class is loadable.

If _not_ loadable, it's possible for its deserialization to fail
leading to the batch dying after only a couple of objects...

adam-vessey committed:
Coding standards.
